projects/storefrontlib/src/lib/checkout/components/multi-step-checkout/container/multi-step-checkout.component.ts

```diff
@@ -45,53 +55,22 @@ export class MultiStepCheckoutComponent implements OnInit, OnDestroy {
     protected cartService: CartService,
     protected cartDataService: CartDataService,
     protected routingService: RoutingService,
+    protected userService: UserService,
     protected globalMessageService: GlobalMessageService,
     protected cd: ChangeDetectorRef
   ) {}
 
-  private refreshCart(): void {
-    this.cartService.loadDetails();
-  }
-
-  ngOnInit() {
-    if (!this.cartDataService.getDetails) {
-      this.cartService.loadDetails();
-    }
-    this.cart$ = this.cartService.getActive();
-    this.processSteps();
+  ngOnInit(): void {
+    this.user$ = this.userService
```

Comment:
I would use here combineLatest with those observables and loadCheckout details here. Would avoid the temp id variables.